willow/app/inputs/object_identifier_input.rb

```diff
@@ -0,0 +1,60 @@
+class ObjectIdentifierInput < NestedAttributesInput
+
+protected
+
+  def build_components(attribute_name, value, index, options)
+    out = ''
+
+    object_identifier_statement = value
+
+    # Inherit required for fields validated in nested attributes
+    required  = false
+    if object.required?(:identifier_nested) and index == 0
+      required = true
+    end
+
+    # --- obj_id_scheme
+    field = :obj_id_scheme
+    field_name = name_for(attribute_name, index, field)
+    field_id = id_for(attribute_name, index, field)
+    field_value = object_identifier_statement.send(field).first
+
+    out << "<div class='row'>"
```

Comment:
Would it be possible to move this html generation code to a rails `.erb` or `.haml` template?, and then call with render('template_name') etc ?

If not easy, its no biggy.

Reply:
I know right. I too found this odd. I've extended the Hyrax code base, but stuck to their style. Hyrax has defined all input fields in app/inputs in .rb files. I would say let's leave it for now.

Reply:
👍  ok sure